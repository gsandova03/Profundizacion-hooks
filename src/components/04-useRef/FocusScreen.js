import React, { useRef } from 'react'

export const FocusScreen = () => {

  const inputRef = useRef();

  const handleClick = () => {
    inputRef.current.select();
  }

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr/>

      <input 
        ref={ inputRef }
        type="text" 
        className="form-control" 
        placeholder="caja de texto"/>

      <button className="btn btn-primary mt-3" onClick={ handleClick }>Focus</button>
    </div>
  )
}
