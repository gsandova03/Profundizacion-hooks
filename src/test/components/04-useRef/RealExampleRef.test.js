import React from 'react'
import { shallow } from "enzyme"
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef'

describe('Pruebas en el componente <RealExampleRef />', () => {
  
  const wrapper = shallow( <RealExampleRef/> )

  test('debe mostrarse correctamente', () => {
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( false );
  })

  test('debe de mostrar el componente <MultipleCustomHooks />', () => {
    const button = wrapper.find( 'button' )

    //usar simulate en un boton para verificar la renderizacion de un componente condicional
    button.simulate('click')

    expect( wrapper.find('MultipleCustomHooks').exists() ).toBe( true )

  })
  
})
