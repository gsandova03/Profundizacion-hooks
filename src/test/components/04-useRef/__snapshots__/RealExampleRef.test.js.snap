// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <RealExampleRef /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <hr />
  <button
    className="btn btn-primary m-3"
    onClick={[Function]}
  >
     Show/Hide 
  </button>
</div>
`;
